src/client/Microsoft.Identity.Client/WwwAuthenticateParameters.cs

```diff
@@ -50,6 +58,17 @@ public class WwwAuthenticateParameters
         /// </summary>
         public string Error { get; set; }
 
+        /// <summary>
+        /// AuthScheme.
+        /// See https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/WWW-Authenticate#syntax for more details
+        /// </summary>
+        public string AuthScheme { get; private set; }
+
+        /// <summary>
+        /// Pop Nonce. This is acquired from with the POP WWW-Authenticate header.
+        /// </summary>
+        public string PopNonce { get; private set; }
```

Comment:
`PopNonce` implies that these WWW-Auth params are tied to the pop auth scheme.

Reply:
I thought you suggested to change this to PopNonce?

Reply:
Sorry, I did not explain this better. 

WWWAuthenticate header is associated with an auth scheme and each scheme can have a nonce. So there can be a nonce param for the POP WWWAuthhenticate header, but at the same time a nonce for the Bearer WWWAuthenticate header or for Basic etc. So in the context of `WWWAuthhenticate`, we cannot hardcode the name "pop" anywhere, as the header is tied to a scheme. It would be a bad api to have "POPNonce" property on the "Bearer" WWWAuth header.

However, the AuthenticationHeader class is a higher level parser. It aims to extract key information from a collection of WWWAuthenticate and from an AuthenticateInfo header. In higher level parser we can have `POPNonce`, and its meaning is "find the WWWAUThenticate header associated with POP scheme and extract the nonce value OR get the AUthenticateInfo nonce".

Hope this helps.